### PR TITLE
Add BandChain Support

### PR DESCRIPTION
**Pull request by ravidsrk**
## Description

Add support for the [BandChain](https://github.com/bandprotocol/bandchain) blockchain . This PR supports sending funds and staking.

Since bandchain uses the Cosmos SDK, the cosmos signing logic is reused. The integration is very similar to the Terra and Kava integration.

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: swift/Tests/Blockchains/BandChainTests.swift [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+    func testAddress() {
+        let address = CoinType.bandChain.deriveAddress(privateKey: privateKey)
+
+        XCTAssertEqual(address, "band1jf9aaj9myrzsnmpdr7twecnaftzmku2mgms4n3")
+        XCTAssertTrue(CoinType.bandChain.validate(address: "band1pnndgfwsrff86263xzpc5cd3t6yfvgjyqc87jh"))
+        XCTAssertTrue(CoinType.bandChain.validate(address: "band1jf9aaj9myrzsnmpdr7twecnaftzmku2mgms4n3"))
+        XCTAssertFalse(CoinType.bandChain.validate(address: "cosmos1hsk6jryyqjfhp5dhc55tc9jtckygx0eph6dd02"))
+    }
+
+    func testSigningTransaction() {
```
Comment (hewigovens) [inline swift/Tests/Blockchains/BandChainTests.swift:22]: can you add some tx signing tests to android as well?

File: coins.json [hunk 2]
```diff
@@ -1380,5 +1380,29 @@
...
+        "name": "BandChain",
+        "symbol": "BAND",
+        "decimals": 6,
+        "blockchain": "Cosmos",
+        "derivationPath": "m/44'/494'/0'/0/0",
+        "curve": "secp256k1",
+        "publicKeyType": "secp256k1",
+        "hrp": "band",
+        "explorer": {
+            "url": "https://scan-wenchang-testnet2.bandchain.org/",
```
[the remark this answers is not in the document]
Comment (ravidsrk) [reply coins.json:1395]: @hewigovens MainNet explorer is `https://cosmoscan.io/` which isn't live yet. This is going live on 10th June. I was thinking of submitting another PR once the MainNet explorer goes live.

File: swift/Tests/Blockchains/BandChainTests.swift [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+import TrustWalletCore
+import XCTest
+
+class BandChainTests: XCTestCase {
+    let privateKey = PrivateKey(data: Data(hexString: "1037f828ca313f4c9e120316e8e9ff25e17f07fe66ba557d5bc5e2eeb7cba8f6")!)!
+
+    func testAddress() {
+        let address = CoinType.bandChain.deriveAddress(privateKey: privateKey)
+
+        XCTAssertEqual(address, "band1jf9aaj9myrzsnmpdr7twecnaftzmku2mgms4n3")
```
[the remark this answers is not in the document]
Comment (ravidsrk) [reply swift/Tests/Blockchains/BandChainTests.swift:16]: `band1jf9aaj9myrzsnmpdr7twecnaftzmku2mgms4n3` is the correct one on both Android and iOS. I have resolved it on both the places now.

File: swift/Tests/Blockchains/BandChainTests.swift (hunk 1 above)
Comment (ravidsrk) [reply swift/Tests/Blockchains/BandChainTests.swift:22]: @hewigovens Added tx signing tests to android as well.

File: docs/coins.md [hunk 4]
```diff
@@ -42,6 +42,7 @@ This list is generated from [./coins.json](../coins.json)
 | 457     | Aeternity        | AE     | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/aeternity/info/logo.png" width="32" />    | <https://aeternity.com>       |
 | 459     | Kava             | KAVA   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/kava/info/logo.png" width="32" />         | <https://kava.io>             |
 | 461     | Filecoin         | FIL    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/filecoin/info/logo.png" width="32" />     | <https://filecoin.io/>        |
+| 494     | BandChain        | BAND   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/band/info/logo.png" width="32" />         | <https://bandprotocol.com/>   |
```
[the remark this answers is not in the document]
Comment (ravidsrk) [reply docs/coins.md:45]: I have created the PR: https://github.com/trustwallet/assets/pull/2297

**Comment by ravidsrk** [comment]
@hewigovens Resolved all the outstanding issues.

**Pull request merged**